src/DonorProfiles/resources/js/app/components/donation-table/index.js

```diff
@@ -0,0 +1,78 @@
+import { Fragment, useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import Table from '../table';
+import DonationRow from './donation-row';
+
+import './style.scss';
+
+const DonationTable = ( { donations, perPage } ) => {
+	const [ page, setPage ] = useState( 1 );
+
+	let donationRows = [];
+	let start = 0;
+	let end = perPage;
+	let lastPage = 1;
+
+	if ( donations ) {
+		start = ( page - 1 ) * perPage;
+		end = start + perPage <= Object.entries( donations ).length ? start + perPage : Object.entries( donations ).length;
```

Comment:
I would suggest extracting this complicated ternary to an object method.

Reply:
Good thinking - went ahead and extracted logic for determining the start index as well: https://github.com/impress-org/givewp/pull/5455/commits/4d6d9c0d763fc7a815c1eab08869169cd24d9873